pact/Pact/Core/IR/Eval/CoreBuiltin.hs

```diff
@@ -1218,9 +1263,14 @@ coreIntToStr info b cont handler _env = \case
     | v < 0 ->
       returnCEK cont handler (VError "int-to-str error: cannot show negative integer" info)
     | base >= 2 && base <= 16 -> do
+      let strLen = 1 + Exts.I# (IntLog.integerLogBase# base v)
+      chargeGasArgs info $ GConcat $ TextConcat $ GasTextLength $ fromIntegral strLen
```

Comment:
Should this be `TextConcat`?

Reply:
Yep, it more or less corresponds to the semantics of `TextConcat` (and is reasonably close to the benchmarked results), which is
```
data ConcatType
  = TextConcat !GasTextLength
  -- ^ Total final string length
...
```

I agree the `GConcat`/`TextConcat` name is a bit overstretched (as I believe one of the comments somewhere in the code mentions), but we can rename that separately.